gnomad_qc/v5/sample_qc/create_sample_qc_metadata_ht.py

```diff
@@ -0,0 +1,634 @@
+) -> hl.Table:
+    """
+    Get relatedness filtering Table for the combined meta Table.
+
+    Add the following related filter boolean annotations to the input `ht` under a
+    `relatedness_filters` struct:
+
+        - related: Whether the sample was filtered for second-degree (or closer)
+          relatedness in the ancestry inference PCA.
+        - duplicate_or_twin: Whether the filtered sample has a duplicate or twin among
+          all samples that are not hard-filtered.
+        - parent_child: Whether the filtered sample has a parent or child among all
+          samples that are not hard-filtered.
+        - sibling: Whether the filtered sample has a sibling among all samples that are
+          not hard-filtered.
+        - Any sample in `ht` that is hard-filtered will have a missing value for these
+          annotations.
+
+    These related filter annotations are also provided for release filtered samples
+    added to the input `ht` under a `release_relatedness_filters` struct:
+
+        - related: Whether the release filtered sample was filtered for
+          second-degree (or closer) relatedness in the final release.
+        - duplicate_or_twin: Whether the release filtered sample has a
+          duplicate or twin among all samples that are not hard-filtered or
+          outlier-filtered.
+        - parent_child: Whether the release filtered sample has a parent or
+          child among all samples that are not hard-filtered or outlier-filtered.
+        - sibling: Whether the release filtered sample has a sibling among all
+          samples that are not hard-filtered or outlier-filtered.
+        - Any sample in `ht` that is hard-filtered or outlier-filtered will have a
+          missing value for these annotations.
+
+    :param ht: Sample QC filter Table to add relatedness filter annotations to.
+    :param relationship_ht: Table with relationships annotations.
+    :param hard_filtered_expr: Boolean Expression indicating whether the sample was
+        hard filtered.
+    :param outlier_filtered_expr: Boolean Expression indicating whether the sample was
+        outlier filtered.
+    :return: Table with related filters added and Table with relationship and gnomad v3
+        overlap information.
+    """
+    rel_dict = {
+        "related": SECOND_DEGREE_RELATIVES,
+        "duplicate_or_twin": DUPLICATE_OR_TWINS,
+        "parent_child": PARENT_CHILD,
+        "sibling": SIBLINGS,
+    }
+    relationships = relationship_ht[ht.key]
+    relatedness_filters_ht = ht.annotate(
+        relatedness_filters=hl.struct(
+            **{
+                rel: get_relationship_filter_expr(
+                    hard_filtered_expr,
+                    hl.is_defined(related_samples_to_drop(release=False).ht()[ht.key]),
+                    relationships.relationships,
+                    rel_val,
+                )
+                for rel, rel_val in rel_dict.items()
+            }
```

Comment:
It doesnt look like these are used downstream, should we drop them here?

Reply:
not sure if this is ever looked at, i can drop